tests/unit/model/test_model.py

```diff
@@ -293,6 +296,65 @@ def test_feature_columns(object_under_test):
     assert object_under_test.feature_columns == expected
 
 
+def test_from_api_repr_w_transform_columns(target_class):
+    resource = {
+        "modelReference": {
+            "projectId": "my-project",
+            "datasetId": "my_dataset",
+            "modelId": "my_model",
+        },
+        "transformColumns": [
+            {
+                "name": "transform_name",
+                "type": {"typeKind": "INT64"},
+                "transformSql": "transform_sql",
+            }
+        ],
+    }
+    got = target_class.from_api_repr(resource)
+    assert len(got.transform_columns) == 1
+    transform_column = got.transform_columns[0]
+    assert isinstance(transform_column, google.cloud.bigquery.model.TransformColumn)
+    assert transform_column.name == "transform_name"
+
+
+def test_transform_column_name():
+    transform_columns = google.cloud.bigquery.model.TransformColumn(
+        {"name": "is_female"}
+    )
+    assert transform_columns.name == "is_female"
+
+
+def test_transform_column_transform_sql():
+    transform_columns = google.cloud.bigquery.model.TransformColumn(
+        {"transformSql": "is_female"}
+    )
+    assert transform_columns.transform_sql == "is_female"
+
+
+def test_transform_column_type():
+    transform_columns = google.cloud.bigquery.model.TransformColumn(
+        {"type": {"typeKind": "BOOL"}}
+    )
+    assert transform_columns.type_.type_kind == "BOOL"
+
+
+def test_transform_column_type_none():
+    transform_columns = google.cloud.bigquery.model.TransformColumn({})
+    assert transform_columns.type_ is None
+
+
+def test_transform_column_properties():
+    transform_column = google.cloud.bigquery.model.TransformColumn(
+        {"name": "is_female", "type": {"typeKind": "BOOL"}, "transformSql": "is_female"}
+    )
+    assert transform_column._properties == {
+        "name": "is_female",
+        "type": {"typeKind": "BOOL"},
+        "transformSql": "is_female",
+    }
```

Comment:
Let's have this be our `from_api_repr` test.

```suggestion
def test_transform_column_from_api_repr_with_unknown_properties():
    transform_column = google.cloud.bigquery.model.TransformColumn.from_api_repr(
        {"name": "is_female", "type": {"typeKind": "BOOL"}, "transformSql": "is_female", "test": "one"}
    )
    assert transform_column._properties == {
        "name": "is_female",
        "type": {"typeKind": "BOOL"},
        "transformSql": "is_female",
        "test": "one",
    }
```